ce/api/util.py

```diff
@@ -99,19 +99,29 @@ def mean_datetime(datetimes):
 def validate_cell_method(cell_method):
     return cell_method in ('mean', 'standard_deviation')
 
+def find_matching_cell_methods(cell_methods, target_method):
+    def filter_on_method(cell_method, target_method):
+        pattern = r'time:[a-z\s]*time:\s+{}\s+over\s+(days|years)'.format(target_method)
+        return re.fullmatch(pattern, cell_method)
+
+    if target_method == 'mean':
```

Comment:
This bit desperately needs an explanatory comment as to _why_ mean has this weird special case handling. Since it's for weird historical reasons, it's definitely _not_ something someone could figure out by reading the code and using logic. Explain what it does and why it has to be done to the future programmer who is looking at this code for the first time.

Reply:
Ok good idea, I added a comment